glambie/util/timeseries_helpers.py

```diff
@@ -593,3 +596,115 @@ def get_average_trends_over_new_time_periods(start_dates, end_dates, changes, ne
+    new_start_dates, new_end_dates, new_changes, new_errors, new_hydro, new_sea_level = [], [], [], [], [], []
+
+    for i in range(len(input_dataframe.start_date) - 1):
+        current_start_date = start_dates[i]
+        current_end_date = end_dates[i]
+        new_start_dates.append(current_start_date)
+        new_end_dates.append(current_end_date)
+        new_changes.append(input_dataframe.glacier_change_observed[i])
+        new_errors.append(input_dataframe.glacier_change_uncertainty[i])
+        new_hydro.append(input_dataframe.hydrological_correction_value[i])
+        new_sea_level.append(input_dataframe.sea_level_correction_value[i])
+
+        if date_gaps_in_days[i] > 1:
+            gap_start_date = current_end_date + relativedelta(days=1)
+            gap_end_date = start_dates[i + 1] - relativedelta(days=1)
+            new_start_dates.append(gap_start_date)
+            new_end_dates.append(gap_end_date)
+            new_changes.append(np.nan)
+            new_errors.append(np.nan)
+            new_hydro.append(np.nan)
+            new_sea_level.append(np.nan)
+
+    # Add last row in
+    new_start_dates.append(start_dates[-1])
+    new_end_dates.append(end_dates[-1])
+    new_changes.append(input_dataframe.glacier_change_observed.tolist()[-1])
+    new_errors.append(input_dataframe.glacier_change_uncertainty.tolist()[-1])
+    new_hydro.append(input_dataframe.hydrological_correction_value.tolist()[-1])
+    new_sea_level.append(input_dataframe.sea_level_correction_value.tolist()[-1])
+
+    interpolated_dataframe = pd.DataFrame()
+    interpolated_dataframe['start_date'] = [datetime.datetime.strftime(a, '%d/%m/%Y') for a in new_start_dates]
+    interpolated_dataframe['end_date'] = [datetime.datetime.strftime(a, '%d/%m/%Y') for a in new_end_dates]
+    interpolated_dataframe['glacier_change_observed'] = new_changes
+    interpolated_dataframe['glacier_change_uncertainty'] = new_errors
+    interpolated_dataframe['hydrological_correction_value'] = new_hydro
+    interpolated_dataframe['sea_level_correction_value'] = new_sea_level
+
+    # calculate days covered by each row
+    date_gaps = [
+        datetime.datetime.strptime(interpolated_dataframe['end_date'][i], '%d/%m/%Y') - datetime.datetime.strptime(
+            interpolated_dataframe['start_date'][i], '%d/%m/%Y') for i in range(len(interpolated_dataframe))]
+    date_gaps_in_days = [a.days for a in date_gaps]
+    interpolated_dataframe['days_covered'] = date_gaps_in_days
+
+    # calculate change per day in each row
+    interpolated_dataframe['glacier_change_per_day'] = [
+        a / b for a, b in zip(interpolated_dataframe.glacier_change_observed, interpolated_dataframe.days_covered)]
+    interpolated_dataframe['glacier_change_uncertainty_per_day'] = [
+        a / b for a, b in zip(interpolated_dataframe.glacier_change_uncertainty, interpolated_dataframe.days_covered)]
```

Comment:
is zip required here? columns in dataframes are stored as numpy arrays so you should be able to do:
```suggestion
    # calculate change per day in each row
    interpolated_dataframe['glacier_change_per_day'] = interpolated_dataframe.glacier_change_observed / interpolated_dataframe.days_covered
    interpolated_dataframe['glacier_change_uncertainty_per_day'] = interpolated_dataframe.glacier_change_uncertainty / interpolated_dataframe.days_covered
```

Reply:
Cool, done!